providers/org/datacite/normalizer.py

```diff
@@ -0,0 +1,523 @@
+import re
+import logging
+
+from share.normalize import ctx, tools
+from share.normalize.parsers import Parser
+from share.normalize.utils import format_doi_as_url
+
+
+logger = logging.getLogger(__name__)
+
+THE_REGEX = re.compile(r'(^the\s|\sthe\s)')
+DATE_REGEX = re.compile(r'(^[^a-zA-Z]+$)')
+
+
+class Link(Parser):
+    url = tools.RunPython('format_link', tools.Text(ctx))
+    type = tools.Static('doi')
+
+    def format_link(self, link):
+        return format_doi_as_url(self, link)
+
+
+class AlternateLink(Parser):
+    schema = 'Link'
+
+    url = tools.Text(ctx)
+    type = tools.Static('misc')
+
+
+class RelatedLink(Parser):
+    schema = 'Link'
+
+    url = tools.Text(ctx)
+    type = tools.RunPython('lower', tools.Try(ctx['@relatedIdentifierType']))
+
+    def lower(self, type):
+        if type:
```

Comment:
Unless `'@relatedIdentifierType'` could contain a `None` value, type will always be defined here. `Try` will not execute the `RunPython` if it fails to find `'@relatedIdentifierType'`.